### PR TITLE
Refactor `getTokenData()` to increase readibility

### PR DESCRIPTION
**Pull request by dcts**
@fliebenberg I've refactored the `getTokenData` function to improve readability by reducing unnecessary nesting. I used the "return early" pattern, which is explained well in [this article](https://medium.com/swlh/return-early-pattern-3d18a41bba8).

I've tested the function, and everything still works as expected. Instead of focusing on the diff, I recommend reviewing the full code [here](https://github.com/dcts/radix.meme/blob/4d33e36ef6baf31746152a9c981cad596481ad6e/utils/api-calls.ts#L97-L220).

### CODE REVIEW

File: utils/api-calls.ts [hunk 1]
```diff
@@ -111,100 +111,108 @@ export async function getTokenData(
...
+  );
+  // Validate
+  if (getResourceApiResult.status != 200) {
+    throw new Error(
+      `Problem fetching metadata for token: ${
+        result.address
+      }. API Result: ${JSON.stringify(getResourceApiResult)}`
+    );
+  }
+  // If no items exist, do not continue
```
Comment (fliebenberg) [inline utils/api-calls.ts:178]: Should we not also throw an error in this case as we will not have anything to display for the token.

**Comment by fliebenberg** [comment]
Happy with suggested changes. We just need to think how we deal with any errors when fetching data. Ideally it should not crash the app, but it should display some kind of notification to the client that there was a problem with fetching data.

File: utils/api-calls.ts (hunk 1 above)
Comment (dcts) [reply utils/api-calls.ts:178]: I just refactored without changing any logic. But I do think we should almost never throw an error and fetch as much data as possible. We can simply return no data for some fields and the client can handle displaying errors to the user for all missing data.

**Comment by dcts** [comment]
> Ideally it should not crash the app, but it should display some kind of notification to the client that there was a problem with fetching data

I agree with that approach! Refactored the api call to:
- only crash (throw error) if no data was retrieved at all (as the first API call failed)
- console error if a field was not found or if second api call was not successfull. In that case we return what we have, end keep the other data empty. 
- the client will be responsible to check whether those values exist, and display a message or UI to the user. 
- since this code runs on the server, error logs will be centrally available to us so we can debug by looking at server logs.

@fliebenberg do you mind if I refactor all other functions inside `api-calls` to follow the same style? (no deep nesting + the error flow outlined above?)

File: utils/api-calls.ts (hunk 1 above)
Comment (dcts) [reply utils/api-calls.ts:181]: I adapted the logic, see comment below.

**Pull request merged**